stdlib/public/Concurrency/YieldingContinuation.swift

```diff
@@ -0,0 +1,196 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020-2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+public struct YieldingContinuation<Element, Failure: Error>: Sendable {
+  @_fixed_layout
+  @usableFromInline
+  internal final class Storage: UnsafeSendable {
+    @usableFromInline
+    var continuation: UnsafeContinuation<Element, Error>?
```

Comment:
You could reduce metadata explosion by hoisting this effectively nongeneric storage class out of the generic outer type, so we only need one class object for all instances.

Reply:
hmm good idea; and then store the continuation as a raw crunchy continuation?

Reply:
extracted!